internal/api/schemas/schemas.go

```diff
@@ -178,15 +178,40 @@ func (s *sch) ValidateQuery(plugin common.Plugin, queryName string) error {
 	return s.validatePlugin(plugin, "query", queryName, s.queries)
 }
 
-func (s *sch) ValidateGlobalVariable(v modelV1.VariableSpec) error {
-	if v.Kind != variable.KindList {
-		return nil
+func (s *sch) ValidateVariableSpec(v modelV1.VariableSpec) error {
+	switch v.Kind {
+	case variable.KindDatasource:
+		return s.validateDatasourceVariable(v)
+	case variable.KindList:
+		return s.validateListVariable(v)
+	case variable.KindText:
+		return nil // no extra validation needed here
+	default:
+		return errors.New("unsupported variable kind")
+	}
+}
+
+func (s *sch) validateDatasourceVariable(v modelV1.VariableSpec) error {
+	datasourceVariableSpec, ok := v.Spec.(*variable.DatasourceSpec)
+	if !ok {
+		return errors.New("Error converting Variable to DatasourceVariable")
+	}
+
+	schemasMap := *s.dts.schemas.Load()
+	for kind := range schemasMap {
```

Comment:
probably you will need to check if `schemasMap == nil` before using it

Reply:
Well in that case it wont iterate and go straight to error "no plugin found" after, I think that's enough?

Reply:
Actually you need to verify before, because when using `*s.dts.schemas.Load()` you accessing to the value of the pointer immediately.

But if the pointer is nil, you will have a panic issue here.

You can try this code :

```go
package main

import (
	"fmt"
	"sync/atomic"
)

func main() {
	var schemas atomic.Pointer[map[string]string]
	testMap := *schemas.Load()
	for k := range testMap {
		fmt.Println(k)
	}
}
```

What you need to do is the following code here:

```go
schemasMap := s.dts.schemas.Load()
if schemasMap == nil {
 // do something probably return an error
}
	
for kind := range *schemasMap {
}
```